### PR TITLE
chore: optimize logger code readability

### PR DESCRIPTION
**Pull request by ylmin**
# What :computer: 
- Optimize the code to improve scalability and readability.

# Why :hand:
- Improve code readability and extensibility, making it easier for developers to understand and expand.

# Evidence :camera:
image

### CODE REVIEW

File: src/utils/logger.ts [hunk 1]
```diff
@@ -1,24 +1,26 @@
-import chalk from "chalk";
+import chalk, { Chalk } from "chalk";
 import { format, createLogger, transports } from "winston";
 
 import { hasColor } from "./helpers.js";
 
 export const errorSymbol = "ⓘ"; // used in ../test-utils/matchers.ts to check for errors in console output
 
+export type LogLevelToColorMap = Record<string, Chalk | ((msg: string) => string)>;
```
Comment (JackHamer09) [inline src/utils/logger.ts:8]: I don't think we need to assign the type separately and also export it. Rest looks good!

File: src/utils/logger.ts [hunk 2]
```diff
@@ -1,24 +1,26 @@
...
+
+const logLevelToColorMap: LogLevelToColorMap = {
```
Comment (JackHamer09) [inline src/utils/logger.ts:10]: Maybe also name it like "logLevelFormatter"

**Review by JackHamer09 — CHANGES_REQUESTED** [review]

File: src/utils/logger.ts (hunk 2 above)
Comment (ylmin) [reply src/utils/logger.ts]: Ok, corrected

**Comment by JackHamer09** [comment]
@ylmin please update the PR according to latest comments

**Comment by ylmin** [comment]
> @ylmin please update the PR according to latest comments

Done

File: src/utils/logger.ts [hunk 3]
```diff
@@ -1,24 +1,24 @@
-import chalk from "chalk";
+import chalk, { Chalk } from "chalk";
 import { format, createLogger, transports } from "winston";
 
 import { hasColor } from "./helpers.js";
```
Comment (JackHamer09) [inline src/utils/logger.ts:4]: Also please make sure you import types as actual types.
image

```suggestion
import chalk from "chalk";
import { format, createLogger, transports } from "winston";

import { hasColor } from "./helpers.js";

import type { Chalk } from "chalk";
```

**Review by JackHamer09 — CHANGES_REQUESTED** [review]

**Review by JackHamer09 — APPROVED** [review]

**Pull request merged**